tfx/utils/kube_utils.py

```diff
@@ -187,3 +244,65 @@ def get_current_kfp_pod(client: k8s_client.CoreV1Api) -> k8s_client.V1Pod:
     return client.read_namespaced_pod(name=pod_name, namespace=namespace)
   except KeyError:
     raise RuntimeError('Cannot determine KFP pod from the environment.')
+
+
+def get_pod(core_api: k8s_client.CoreV1Api, pod_name: Text,
+            namespace: Text) -> Optional[k8s_client.V1Pod]:
+  """Get a pod from Kubernetes metadata API.
+  Args:
+    core_api: Client of Core V1 API of Kubernetes API.
+    pod_name: The name of the POD.
+    namespace: The namespace of the POD.
+  Returns:
+    The found POD object. None if it's not found.
+  Raises:
+    RuntimeError: When it sees unexpected errors from Kubernetes API.
+  """
+  try:
+    return core_api.read_namespaced_pod(name=pod_name, namespace=namespace)
+  except k8s_client.rest.ApiException as e:
+    if e.status != 404:
+      raise RuntimeError('Unknown error! \nReason: %s\nBody: %s' %
+                         (e.reason, e.body))
+    return None
+
+
+def wait_pod(core_api: k8s_client.CoreV1Api,
+             pod_name: Text,
+             namespace: Text,
+             exit_condition_lambda: Callable[[k8s_client.V1Pod], bool],
+             condition_description: Text,
+             timeout_sec: int = 300,
+             expotential_backoff: bool = False) -> k8s_client.V1Pod:
+  """Wait for a POD to meet an exit condition.
+  Args:
+    core_api: Client of Core V1 API of Kubernetes API.
+    pod_name: The name of the POD.
+    namespace: The namespace of the POD.
+    exit_condition_lambda: A lambda which will be called intervally to wait
+      for a POD to exit. The function returns True to exit.
+    condition_description: The description of the exit condition which will be
+      set in the error message if the wait times out.
+    timeout_sec: The seconds for the function to wait. Defaults to 300s.
+  Returns:
+    The POD object which meets the exit condition.
+  Raises:
+    RuntimeError: when the function times out.
+  """
+  start_time = datetime.datetime.utcnow()
+  # Exponential backoff: https://cloud.google.com/storage/docs/exponential-backoff
```

Comment:
Why is this particular cloud storage page referenced?

Reply:
The link point to exponential back-off algorithm used here.